bin/hipvars.pm

```diff
@@ -93,10 +93,19 @@ if ($isWindows) {
 # HIP_ROCCLR_HOME is used by Windows builds
 $HIP_ROCCLR_HOME=$ENV{'HIP_ROCCLR_HOME'};
 
+if ($isWindows) {
+    $LIB = "lib";
+} else {
+    # Look for the HSA runtime library to determine the lib directory name
+    $LIB = `find $HIP_PATH/lib $HIP_PATH/lib64 -name libhsa-runtime64* | grep -m1 -o '.*/lib.*/'`;
+    chomp($LIB); # Remove trailing newline characters
+    $LIB = basename($LIB); # Get the lib directory name
```

Comment:
We could change this part to the following in order to allow for the Debian layout to be caught by this as well. However, this has the potential to catch also some other weird cases, and I think it might be better to check on the different cases explicitly. I will push another change soon.
``` perl
    # Look for the AMD HIP library to determine the lib directory name
    $LIB = `find $HIP_PATH/lib $HIP_PATH/lib64 -name libamdhip64* | grep -m1 -o 'lib.*'`;
    chomp($LIB); # Remove trailing newline characters
    $LIB = dirname($LIB); # Get the lib directory name
```

Reply:
I have pushed another commit now which makes the checks more explicit and the code more readable.